dashboard/src/components/TimeRange/GranularitySection.tsx

```diff
@@ -28,24 +31,29 @@ export function GranularitySection({
         {ordered.map((gran) => {
           const isActive = selectedGranularity === gran.value && !disabled;
           const isAllowed = allowedGranularities.includes(gran.value);
+          const showTooltip = !isAllowed && !disabled && notAllowedTitle !== undefined;
           return (
-            <Button
-              key={gran.value}
-              type='button'
-              variant='ghost'
-              size='sm'
-              className={cn(
-                'h-7 w-full cursor-pointer rounded-[4px] border border-transparent px-2 text-[10px] font-medium',
-                !isAllowed && 'cursor-not-allowed opacity-50 hover:opacity-50',
-                !isActive && 'text-muted-foreground hover:bg-accent',
-                isActive && 'bg-input text-foreground border-border hover:bg-input hover:dark:bg-input shadow-sm',
+            <DisabledTooltip key={gran.value} disabled={showTooltip} message={notAllowedTitle ?? ''}>
+              {() => (
+                <Button
+                  type='button'
+                  variant='ghost'
+                  size='sm'
+                  className={cn(
+                    'h-7 w-full cursor-pointer rounded-[4px] border border-transparent px-2 text-[10px] font-medium',
+                    !isAllowed && 'cursor-not-allowed opacity-50 hover:opacity-50',
+                    !isActive && 'text-muted-foreground hover:bg-accent',
+                    isActive &&
+                      'bg-input text-foreground border-border hover:bg-input hover:dark:bg-input shadow-sm',
```

Comment:
```suggestion
                    isActive ? 'bg-input text-foreground border-border hover:bg-input hover:dark:bg-input shadow-sm' : 'text-muted-foreground hover:bg-accent'
```

Code reads easier if you use a ternary here